### DISCUSSION TITLE
append username to folder when creating a virtual folder mapping

### DISCUSSION THREAD

**Discussion by nader-eloshaiker** [drakkan/sftpgo#929] [Ideas]
When adding a virtual folder to a user, you specify the virtual folder path and then the folder name to map to.
It would be great if a `user name` can be appended to the folder path in the creation of the virtual path mapping.

Currently this is what I do:
Docker volume mappings:
/srv/sftpgo/data --> /user (usernames in this folder)
/srv/sftpgo/virtual/photos --> /share/photos (usernames in this folder)

folders: (implemented for each user)
- name: photos-username1 - path: /srv/sftpgo/virtual/photos/username1

users:
- user home path is /srv/sftpgo/data/username1
virtual path:
- vpath: /photos - folder name: photos-username1

The issue here is that I have to create the photos folders for each user and then map each one in the virtual folder.

What I would like to do is:

folders: (one folder)
- name: photos - path: /srv/sftpgo/virtual/photos

users:
- user home path is /srv/sftpgo/data/username1
virtual path:
- vpath: /photos - folder name: `photos/${username}`

**Comment by drakkan** [comment]
Hi, 

this feature is already supported, just read the doc [here](https://github.com/drakkan/sftpgo/blob/main/docs/groups.md). It looks like you have missed [this](https://github.com/drakkan/sftpgo#sponsors) too

**Reply by nader-eloshaiker** [reply]
I did see that but it doesn’t work per the use case I provided as I currently understand it.

I require %username% on the folder path and not on the folder vpath, for the photos folder definition/mapping.

**Reply by drakkan** [reply]
> I did see that but it doesn’t work per the use case I provided as I currently understand it.
> 
> I require %username% on the folder path and not on the folder vpath, for the photos folder definition/mapping.

it is supported

**Reply by nader-eloshaiker** [reply]
Apologies, I made an assumption around what was allowed and not allowed on the folders page of the website given that the `%username%` banner was only presented on the groups page.
